tensorflow/core/common_runtime/threadpool_device.cc

```diff
@@ -43,7 +47,26 @@ ThreadPoolDevice::ThreadPoolDevice(const SessionOptions& options,
     : LocalDevice(options, Device::BuildDeviceAttributes(
                                name, DEVICE_CPU, memory_limit, locality)),
       allocator_(allocator),
-      scoped_allocator_mgr_(new ScopedAllocatorMgr(name)) {}
+      scoped_allocator_mgr_(new ScopedAllocatorMgr(name)) {
+#ifdef INTEL_MKL
+#ifdef _OPENMP
+  const char* user_omp_threads = getenv("OMP_NUM_THREADS");
```

Comment:
std::getenv is only thread-safe with other getenv but not setenv and unsetenv. Some other parts of TensorFlow use setenv and unsetenv, like Jayaram said. Just a note that we might want to look into this in the future.

Reply:
As long as {set,unset,put,clear}env only happen in tests, I think this is fine: even if the tests are not single-threaded, the only damage that can be done is to the tests themselves.

Reply:
Oops. Sorry. For some reason I thought core/platform/posix/subprocess.cc is using setenv, but it actually just mentions setenv. Just confirmed that there aren't any {set,unset,put,clear}env calls outside of tests. That's great! :)